app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -121,24 +122,37 @@ public final void setSynapseClient(SynapseClient synapseClient) {
         this.synapseClient = synapseClient;
     }
 
-    public Study getStudy(String identifier) {
+    private Study getStudy(String identifier, boolean containDeactivated) {
         checkArgument(isNotBlank(identifier), Validate.CANNOT_BE_BLANK, "identifier");
-        
+
         Study study = cacheProvider.getStudy(identifier);
         if (study == null) {
             study = studyDao.getStudy(identifier);
             cacheProvider.setStudy(study);
         }
+
+        if (study != null && !study.isActive() && !containDeactivated) {
+            throw new EntityNotFoundException(Study.class, "Study '"+identifier+"' is de-activated.");
+        }
```

Comment:
I think the logic is wrong... if study is null, we DO thrown ENFE, don't we? study.isActive/containDeactivated are paired relative to that:

(study == null || (!study.isActive() && !containDeactivated))